metron-platform/metron-writer/src/main/java/org/apache/metron/writer/FlushPolicy.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.writer;
+
+import org.apache.metron.common.configuration.writer.WriterConfiguration;
+import org.apache.metron.common.writer.BulkMessageWriter;
+import org.apache.metron.common.writer.BulkWriterMessage;
+
+import java.util.List;
+
+/**
+ * This interface is used by the {@link org.apache.metron.writer.BulkWriterComponent} to determine if a batch should be flushed.
+ * @param <MESSAGE_T> Message type
+ */
```

Comment:
Can I ask how the MESSAGE_T is relevant to this interface?  Don't you just need the size?  Why does this need this?

Reply:
Yeah I guess you're right.  My thinking was we might want a policy that depends on message contents at some point.  We don't have an actual use for that now so it is unnecessary.  If we do have a need in the future we can just change the interface since it's internal.

Reply:
Well, there is nothing about the function of the interface ( the call itself ) that needs the type.  You can always implement the interface with an implementing class that is specialized.

Reply:
I think your initial reaction was correct.  Let's keep it simple for now instead of trying to predict the future.  Latest commit uses batch size instead.